src/Auth/Storage/StorageInterface.php

```diff
@@ -30,10 +30,10 @@ public function read();
     /**
      * Write user record.
      *
-     * @param array $user User record.
+     * @param array|\ArrayAccess $user User record.
      * @return void
      */
-    public function write(array $user);
+    public function write($user);
```

Comment:
Adding a new interface won't help if someone has for example extended `SessionStorage` and overridden its `write()` method.

@lorenzo I would like to share your dislike :)